p2p/p2p-core/src/types.rs

```diff
@@ -24,100 +25,48 @@ pub struct BanState<A: NetZoneAddress> {
     pub unban_instant: Option<Instant>,
 }
 
-/// An enumeration of address types.
-///
-/// Used [`ConnectionInfo::address_type`].
-#[derive(Copy, Clone, Default, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
-#[repr(u8)]
-pub enum AddressType {
-    #[default]
-    Invalid,
-    Ipv4,
-    Ipv6,
-    I2p,
-    Tor,
-}
-
-impl AddressType {
-    /// Convert [`Self`] to a [`u8`].
-    ///
-    /// ```rust
-    /// use cuprate_p2p_core::types::AddressType as A;
-    ///
-    /// assert_eq!(A::Invalid.to_u8(), 0);
-    /// assert_eq!(A::Ipv4.to_u8(), 1);
-    /// assert_eq!(A::Ipv6.to_u8(), 2);
-    /// assert_eq!(A::I2p.to_u8(), 3);
-    /// assert_eq!(A::Tor.to_u8(), 4);
-    /// ```
-    pub const fn to_u8(self) -> u8 {
-        self as u8
-    }
-
-    /// Convert a [`u8`] to a [`Self`].
-    ///
-    /// # Errors
-    /// This returns [`None`] if `u > 4`.
-    ///
-    /// ```rust
-    /// use cuprate_p2p_core::types::AddressType as A;
-    ///
-    /// assert_eq!(A::from_u8(0), Some(A::Invalid));
-    /// assert_eq!(A::from_u8(1), Some(A::Ipv4));
-    /// assert_eq!(A::from_u8(2), Some(A::Ipv6));
-    /// assert_eq!(A::from_u8(3), Some(A::I2p));
-    /// assert_eq!(A::from_u8(4), Some(A::Tor));
-    /// assert_eq!(A::from_u8(5), None);
-    /// ```
-    pub const fn from_u8(u: u8) -> Option<Self> {
-        Some(match u {
-            0 => Self::Invalid,
-            1 => Self::Ipv4,
-            2 => Self::Ipv6,
-            3 => Self::I2p,
-            4 => Self::Tor,
-            _ => return None,
-        })
-    }
-}
-
-// TODO: reduce fields and map to RPC type.
-//
 /// Data within [`crate::services::AddressBookResponse::ConnectionInfo`].
 pub struct ConnectionInfo<A: NetZoneAddress> {
+    // The following fields are mostly the same as `monerod`.
     pub address: A,
     pub address_type: AddressType,
     pub avg_download: u64,
     pub avg_upload: u64,
-    pub connection_id: u64, // TODO: boost::uuids::uuid
     pub current_download: u64,
     pub current_upload: u64,
     pub height: u64,
+    /// Either a domain or an IP without the port.
     pub host: String,
     pub incoming: bool,
-    pub ip: String,
     pub live_time: u64,
     pub localhost: bool,
     pub local_ip: bool,
-    pub peer_id: String,
-    pub port: String,
+    pub peer_id: u64,
     pub pruning_seed: PruningSeed,
     pub recv_count: u64,
     pub recv_idle_time: u64,
     pub rpc_credits_per_hash: u32,
     pub rpc_port: u16,
     pub send_count: u64,
     pub send_idle_time: u64,
-    pub state: String, // TODO: what type is this?
+    pub state: ConnectionState,
     pub support_flags: u32,
+
+    // The following fields are slightly different than `monerod`.
+    //
+    /// [`None`] if Tor/i2p or unknown.
+    pub socket_addr: Option<std::net::SocketAddr>,
+    // This field does not exist for Cuprate:
+    // <https://github.com/Cuprate/cuprate/pull/320#discussion_r1811335020>
+    // pub connection_id: u128,
```

Comment:
I guess we'll just use `FIELD_NOT_SUPPORTED` for fields like this?

Thankfully the RPC type is a hex `String` - I guess for types like `u64` we just use `Default::default()`?

Reply:
I think it might be best to try mimic the format but just set it to a default value, like `connection_id` is a UUID right? we should just set it to a default UUID and hex encode that.

Reply:
I agree with @Boog900 . If I had to debug breakage in migration i would prefer it to be still a default id of some sort rather an another type altogether. Just to avoid a big gasp here.

Reply:
Ah ok, I thought this would be returned in RPC requests, if this is just internal and we do replace it with data that decodes into the correct type before sending it over the wire then I am ok with that.